onnxruntime/contrib_ops/cpu/tokenizer.cc

```diff
@@ -365,77 +192,85 @@ Status Tokenizer::CharTokenize(OpKernelContext* ctx, size_t N, size_t C,
 Status Tokenizer::SeparatorTokenize(OpKernelContext* ctx,
                                     size_t N, size_t C,
                                     const std::vector<int64_t>& input_dims) const {
-  struct Match {
-    int priority_;
-    size_t offset_;
-    size_t size_;
-    // create a conflict for overlapping matches
-    // thus if they overlap neither is less than the other
-    // and they are considered equal
-    bool operator<(const Match& o) const {
-      return (offset_ + size_) <= o.offset_;
-    }
-  };
+  using namespace re2;
+  std::vector<std::vector<StringPiece>> rows;
+  rows.reserve(N * C);
+
+  // We do not constraint the search to match
+  // on the beginning or end of the string
+  const RE2::Anchor anchor = RE2::UNANCHORED;
```

Comment:
I assume that the matching process would still try the find tokens from the beginning of a string, right?

Reply:
The search start/end is controlled by start_post/end_pos. The mode we are using makes this argument not relevant.

---
In reply to: [276862868](https://github.com/Microsoft/onnxruntime/pull/857#discussion_r276862868) [](ancestors = 276862868)